### DISCUSSION TITLE
How can I add a npm package to Directus running in Docker

### DISCUSSION THREAD

**Discussion by benjibenoit** [directus/directus#14621] [Archive]
### Describe the Bug

I need to add a package to my Directus for professional purposes. I have retrieved the package.json from the root image, added my package but Directus never npm i. Do you have a solution so that I can add new packages? 

### To Reproduce

Create a Directus via Docker. Mount the package.json and try to add a new package

### Errors Shown

09:02:40 ⚠️  Couldn't load extensions
09:02:40 ⚠️  Cannot find module 'XXXXXXXXX/package.json'
Require stack:
- /directus/node_modules/@directus/shared/dist/cjs/utils/node/resolve-package.js
- /directus/node_modules/@directus/shared/dist/cjs/utils/node/get-extensions.js
- /directus/node_modules/@directus/shared/dist/cjs/utils/node/index.js
- /directus/node_modules/directus/dist/extensions.js
- /directus/node_modules/directus/dist/services/graphql.js
- /directus/node_modules/directus/dist/services/index.js
- /directus/node_modules/directus/dist/middleware/respond.js
- /directus/node_modules/directus/dist/controllers/activity.js
- /directus/node_modules/directus/dist/app.js
- /directus/node_modules/directus/dist/server.js
- /directus/node_modules/directus/dist/cli/index.js
- /directus/node_modules/directus/dist/cli/run.js
- /directus/node_modules/directus/cli.js

### What version of Directus are you using?

9.14

### What version of Node.js are you using?

16

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by br41nslug** [comment]
A docker image is already a built version of the software so you can't really add packages after the build process. Well of course you can open a shell in the docker and install a package that way with just `npm i package` but because docker images are read-only that will be gone when you restart the docker. Usually you wouldnt add packages to the root image but compile them with the extension you're deploying.
You can add pacakges to the image by extending it using a custom image like this:
```Dockerfile
FROM directus/directus:9.14.5

WORKDIR /directus

RUN npm install somepackage someOtherPackage
```
https://docs.docker.com/get-started/09_image_best/

**Reply by benjibenoit** [reply]
I tryed your code with my needs and I got this error :
![image](https://user-images.githubusercontent.com/32774804/180743655-7713d73a-cbc3-466d-999f-667a8b7d9b66.png)

**Reply by br41nslug** [reply]
which of the 2 options i proposed did you use? installing a single (or multiple) packages using `npm install package` or upload a new `package.json` with `npm install`?

**Reply by br41nslug** [reply]
Probably best to stick to the first form when building custom images as i can verify that works 😄

**Reply by benjibenoit** [reply]
I tried with package.json
I'll try with your idea asap 😉

**Reply by benjibenoit** [reply]
> Probably best to stick to the first form when building custom images as i can verify that works 😄

It seems to work like that! Thanks for the tip
